to_xe_bytes for isize and usize returns an array of different size

stepancheg opened the pull request:
... on different platforms.

Official rustdoc of
[`usize::to_le_bytes`](https://doc.rust-lang.org/std/primitive.usize.html#method.to_le_bytes)
displays signature

```
pub fn to_ne_bytes(self) -> [u8; 8]
```

which might be misleading: this function returns 4 bytes on 32-bit
systems.

With this commit applied rustdoc for `isize` and `usize` is this:
<img width="740" alt="2019-04-15_0020" src="https://user-images.githubusercontent.com/28969/56100765-9f69b380-5f14-11e9-974c-daa25edaa881.png">

rust-highfive commented:
r? @bluss

(rust_highfive has picked a reviewer for you, use r? to override)

NieDzejkob commented on src/libcore/num/mod.rs:
```diff
@@ -212,11 +212,20 @@ pub mod diy_float;
+macro_rules! usize_isize_to_xe_bytes_doc {
+    () => {"
+
+This function returns an array of length 2, 4 or 8 bytes depending
```
It might be a good idea to add a bold `**NOTE**:` -- a (partially) wrong function signature being shown is pretty important.

NieDzejkob commented on src/libcore/num/mod.rs:
```diff
@@ -212,11 +212,20 @@ pub mod diy_float;
+    () => {"
+
+This function returns an array of length 2, 4 or 8 bytes depending
+on target pointer size.
```
While I'm not a native English speaker, I feel like there should be a `the` before `target`.

bors commented:
:umbrella: The latest upstream changes (presumably #60013) made this pull request unmergeable. Please resolve the merge conflicts.

stepancheg commented:
Rebased and added the article. Looks like this now:

<img width="716" alt="2019-04-24_0027" src="https://user-images.githubusercontent.com/28969/56622453-1fd09880-6628-11e9-9591-56af13ad690d.png">

Centril commented:
r? @ehuss

Centril commented:
@bors delegate=ehuss

bors commented:
:v: @ehuss can now approve this pull request

ehuss commented:
Can a similar treatment be done for `from_*_bytes` which I believe also take different sizes?

stepancheg commented:
Updated PR with `from_*_bytes` docs.

rust-highfive commented:
The job `x86_64-gnu-llvm-6.0` of your PR [failed on Travis](https://travis-ci.com/rust-lang/rust/jobs/197782267) ([raw log](https://api.travis-ci.com/v3/job/197782267/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
travis_time:end:04a3f5dc:start=1557088832296510136,finish=1557088833116044029,duration=819533893
$ git checkout -qf FETCH_HEAD
travis_fold:end:git.checkout

Encrypted environment variables have been removed for security reasons.
See https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
$ export SCCACHE_BUCKET=rust-lang-ci-sccache2
$ export SCCACHE_REGION=us-west-1
$ export GCP_CACHE_BUCKET=rust-lang-ci-cache
$ export AWS_ACCESS_KEY_ID=AKIA46X5W6CZEJZ6XT55
---

[00:04:32] travis_fold:start:tidy
travis_time:start:tidy
tidy check
[00:04:32] tidy error: /checkout/src/libcore/num/mod.rs:240: line longer than 100 chars
[00:04:32] tidy error: /checkout/src/libcore/num/mod.rs:2294: line longer than 100 chars
[00:04:37] some tidy checks failed
[00:04:37] 
[00:04:37] 
[00:04:37] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/tidy" "/checkout/src" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "--no-vendor" "--quiet"
[00:04:37] 
[00:04:37] 
[00:04:37] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test src/tools/tidy
[00:04:37] Build completed unsuccessfully in 0:01:06
[00:04:37] Build completed unsuccessfully in 0:01:06
[00:04:37] Makefile:67: recipe for target 'tidy' failed
[00:04:37] make: *** [tidy] Error 1
The command "stamp sh -x -c "$RUN_SCRIPT"" exited with 2.
travis_time:start:00e626b8
$ date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
Sun May  5 20:45:21 UTC 2019
---
travis_time:end:03f61b54:start=1557089122744413341,finish=1557089122749326561,duration=4913220
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:0e305d00
$ ln -s . checkout && for CORE in obj/cores/core.*; do EXE=$(echo $CORE | sed 's|obj/cores/core\.[0-9]*\.!checkout!\(.*\)|\1|;y|!|/|'); if [ -f "$EXE" ]; then printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" "$CORE"; gdb --batch -q -c "$CORE" "$EXE" -iex 'set auto-load off' -iex 'dir src/' -iex 'set sysroot .' -ex bt -ex q; echo travis_fold":"end:crashlog; fi; done || true
travis_fold:end:after_failure.4
travis_fold:start:after_failure.5
travis_time:start:11f5b3b4
travis_time:start:11f5b3b4
$ cat ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers || true
cat: ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers: No such file or directory
travis_fold:end:after_failure.5
travis_fold:start:after_failure.6
travis_time:start:00814a69
$ dmesg | grep -i kill
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

ehuss commented:
Thank you!

@bors r+ rollup

bors commented:
:pushpin: Commit 764e3668b05eeee92e8d81ac748cbb1c76de1a44 has been approved by `ehuss`

<!-- @bors r=ehuss 764e3668b05eeee92e8d81ac748cbb1c76de1a44 -->

bors commented:
:hourglass: Testing commit 764e3668b05eeee92e8d81ac748cbb1c76de1a44 with merge 3b68ed23f1364433b0d6a7f652ae73a42b38a10a...
<!-- homu: {"type":"BuildStarted","head_sha":"764e3668b05eeee92e8d81ac748cbb1c76de1a44","merge_sha":"3b68ed23f1364433b0d6a7f652ae73a42b38a10a"} -->

bors commented:
:broken_heart: Test failed - [checks-travis](https://travis-ci.com/rust-lang/rust/builds/110987976)
<!-- homu: {"type":"BuildFailed","builder_url":"https://travis-ci.com/rust-lang/rust/builds/110987976","builder_name":"checks-travis"} -->

rust-highfive commented:
The job `x86_64-gnu-aux` of your PR [failed on Travis](https://travis-ci.com/rust-lang/rust/jobs/198443483) ([raw log](https://api.travis-ci.com/v3/job/198443483/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
[02:12:49] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/cargotest" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "/checkout/obj/build/ct"
[02:12:49] expected success, got: exit code: 101
[02:12:49] 
[02:12:49] 
[02:12:49] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test src/test/run-pass/pretty src/test/run-fail/pretty src/test/run-pass-valgrind/pretty src/test/run-pass-fulldeps/pretty src/tools/cargo src/tools/cargotest
[02:12:49] Build completed unsuccessfully in 0:43:43
[02:12:49] make: *** [check-aux] Error 1
[02:12:49] Makefile:50: recipe for target 'check-aux' failed
The command "stamp sh -x -c "$RUN_SCRIPT"" exited with 2.
travis_time:start:09d69d10
$ date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
Wed May  8 05:38:40 UTC 2019
---
travis_time:end:11fbc164:start=1557293924206558359,finish=1557293924214982811,duration=8424452
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:11a4d8e9
$ ln -s . checkout && for CORE in obj/cores/core.*; do EXE=$(echo $CORE | sed 's|obj/cores/core\.[0-9]*\.!checkout!\(.*\)|\1|;y|!|/|'); if [ -f "$EXE" ]; then printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" "$CORE"; gdb --batch -q -c "$CORE" "$EXE" -iex 'set auto-load off' -iex 'dir src/' -iex 'set sysroot .' -ex bt -ex q; echo travis_fold":"end:crashlog; fi; done || true
travis_fold:start:crashlog
obj/cores/core.5500.!checkout!obj!build!x86_64-unknown-linux-gnu!stage2!bin!rustc
[New LWP 5501]
[New LWP 5500]
warning: Could not load shared library symbols for 8 libraries, e.g. /lib/x86_64-linux-gnu/libc.so.6.
Use the "info sharedlibrary" command to see the complete listing.
Do you need "set solib-search-path" or "set sysroot"?
Core was generated by `rustc --crate-name foo src/lib.rs --color never --crate-type lib --emit=dep-inf'.
Program terminated with signal SIGABRT, Aborted.
#0  0x00007f5daceca428 in ?? ()
[Current thread is 1 (LWP 5501)]
#0  0x00007f5daceca428 in ?? ()
#1  0x00007f5dacecc02a in ?? ()
#2  0x0000000000000020 in ?? ()
#3  0x0000000000000000 in ?? ()
travis_time:end:11a4d8e9:start=1557293924219875345,finish=1557293926215863072,duration=1995987727
travis_fold:end:after_failure.4
travis_fold:start:after_failure.5
travis_time:start:1eb0873f
travis_time:start:1eb0873f
$ cat ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers || true
cat: ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers: No such file or directory
travis_fold:end:after_failure.5
travis_fold:start:after_failure.6
travis_time:start:32df34d5
$ dmesg | grep -i kill
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

ehuss commented:
Network error: ```[02:12:48] warning: spurious network error (2 tries remaining): failed to get 200 response from `https://static.crates.io/crates/termcolor/termcolor-1.0.1.crate`, got 504```

@bors retry
